### DISCUSSION TITLE
How to use multiple keys table?

### DISCUSSION THREAD

**Discussion by damody** [rbatis/rbatis#289] [General]
```
create TABLE mastery (
    uid varchar(64) Character SET utf8 Default "",
    hero varchar(32) Character SET utf8 Default "",
    lv TINYINT UNSIGNED NOT NULL,
    exp INT UNSIGNED NOT NULL,
    create_date DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
    PRIMARY KEY (`uid`, `hero`)
);
```
like this table

```

#[crud_enable( id_name:"uid,hero" |  id_type:"String,String" | table_name:"mastery")]
#[derive(Clone, Debug, PartialEq, Default)]
pub struct SqlMastery {
    pub uid: String,
    pub hero: String,
    pub lv: i32,
    pub exp: i32,
    pub create_date: Option<NaiveDateTime>,
}
```
how to support?

**Comment by zhuxiujia** [comment]
> ```
> create TABLE mastery (
>     uid varchar(64) Character SET utf8 Default "",
>     hero varchar(32) Character SET utf8 Default "",
>     lv TINYINT UNSIGNED NOT NULL,
>     exp INT UNSIGNED NOT NULL,
>     create_date DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
>     PRIMARY KEY (`uid`, `hero`)
> );
> ```
> 
> like this table
> 
> ```
> 
> #[crud_enable( id_name:"uid,hero" |  id_type:"String,String" | table_name:"mastery")]
> #[derive(Clone, Debug, PartialEq, Default)]
> pub struct SqlMastery {
>     pub uid: String,
>     pub hero: String,
>     pub lv: i32,
>     pub exp: i32,
>     pub create_date: Option<NaiveDateTime>,
> }
> ```
> 
> how to support?

you can use Wrapper create sql。 The latest version has removed 'id_name' and 'id_type'
